src/package-hoister.js

```diff
@@ -835,6 +835,14 @@ export default class PackageHoister {
       // up of modules from different registries so we need to handle this specially
       const parts: Array<string> = [];
       const keyParts = key.split('#');
+      const isWorkspaceEntry = this.workspaceLayout && keyParts[0] === this.workspaceLayout.virtualManifestName;
+
+      // we don't need to install the virtual manifest or workspace packages
+      // which already have a different version in the root
+      if (isWorkspaceEntry && keyParts.length <= 2) {
```

Comment:
Where is the second part of the condition (`[...] which already have a different version in the root`)?

Reply:
It is dependent on the length:

So basically it is this:

```js
if (
  this.workspaceLayout && // Do we have a workspace?
  keyParts[0] === this.workspaceLayout.virtualManifestName && // Is the first keypart the virtual manifest name? If so we are dealing with modules from within workspaces or the virtual manifest itself
  (
    keyParts.length === 1 || // the key is "workspace-aggregator-1234" Which is the virtual workspace itself
    keyParts.length === 2  //  the key is "workspace-aggregator-1234#child" Which is one of the workspace childs which could not be hoisted.
  )
){}
```

Reply:
I also updated the comment to describe it better.